### ISSUE TITLE
Migrating to configuration version 2

### ISSUE THREAD

**Issue by b5** [qri-io/qri#1346]
This PR should land the _capacity_ to run migration from configuration version 1 to 2, but leave running the migration itself off the default configuration so we can keep working without requiring `master` to run a migration.

The config migration should also handle migrating & updating IPFS deps & repo, but that should be handled elsewhere. This issue is to track migrating our configuration. The IPFS stuff should build be added to a migration function, and defined in `qri-io/qfs/qipfs`

### Steps to complete
* close #1345, adding support for multiaddresses
* Remove Stale configuration fields specified in RFC0029:  https://github.com/qri-io/rfcs/blob/master/text/0029-config_revision.md#remove-stale-configuration-fields
* Add in the `Filesystems` field, which can be passed down to a `muxfs.New`, unblocks the leet issue: #1337
* Test the migration, including a rollback

**Comment by b5** [comment]
from our chat:

```go
package config

import (
	qipfs "github.com/qri-io/qfs/cafs/ipfs"
)


func OneToTwo(qripath string) error {

	// REQUIRE migrating into repo_path/ipfs
	// move IPFS folder inside Qri
	// upgrade IPFS version
	// change IPFS ports
	if err := qipfs.RunMigration(); err != nil {
		return qerr.New(err, "qri encountered an error while updating IPFS, there may be an issue with your repo")
	}
		
		// * copy folder
		// * run migration
		// * change ports

	// migrate qri configuration struct
	//    * use multiaddrs
	//    * remove stale fields
	//    * add filesystem field
	//      - now IPFS_PATH comes from qfs.Filesystem["ipfs"].options.path
	if err := oneToTwo(qripath); err != nil {
		return qerr.New(err, "qri encountered an error while updating your configuration file, please re-try")
	}

}

func oneToTwo(qripath string) error {
	ioutil.WriteFile("config_updated.yaml")
}

// package qipfs
func RunMigration() {
	rollback := func() {
		os.RemoveAll()
	}
		defer func() {
			if rollback != nil {
				rollback()
			}
		}

		rollback = nil
		return nil
}

var yalmConfig = `
filesystems:
- type: ipfs
	options:
		path: ./ipfs
- type: local
- type: http
`

`
QRI_PATH
	config.yaml
	ipfs/
`
```

**Comment by ramfox** [comment]
Clarifying the responsibility of `qfs.RunMigrations`
`func RunMigrations(ipfsRepoPath, newRepoPath) error`
  - gets a previous repo path and new repo path
  - copies the previous repo
  - moves the copy onto the newRepoPath
  - migrates the new repo to the correct version
  - if there are any errors, it removes the new repo and the copy

If there are any errors in `OneToTwo` after `RunMigrations` then it is responsible for removing the new repo. And if there are no errors and we successfully move from version one to two, `OneToTwo` is responsible for removing the original ipfs repo

```
func OneToTwo(cfg Config) error {
        ipfsRepoPath = GetIPFSRepoPathHowever()
        qriPath = GetQriPathHowever()
        newRepoPath = filepath.Join(qriPath, "ipfs")
        rollback := func() {
            os.RemoveAll(newRepoPath)
       }
       defer func () {
           if rollback != nil {
               rollback()
           }
       }
        if err := qipfs.RunMigration(tmpDir, newRepoPath); err != nil {
		return qerr.New(err, "qri encountered an error while updating IPFS, there may be an issue with your repo")
	}

	if err := oneToTwo(qripath); err != nil {
		return qerr.New(err, "qri encountered an error while updating your configuration file, please re-try")
	}
        os.RemoveAll(ipfsRepoPath)
        rollback = nil
        return nil
}
```

Happy to change direction with feedback, just outlining my assumptions for clarity as we all work on this together

**Comment by Arqu** [comment]
I've added a partial implementation as a separate PR to align on the implementation details before we all commit to it.

https://github.com/qri-io/qri/pull/1367/files

**Comment by Arqu** [comment]
To elaborate a bit on the changes to config - since merging https://github.com/qri-io/qri/pull/1370 config no longer errors out on unknown params in the config. This allows us to easily delete/ignore fields that no longer exist in the Go config structs. 

However, this can introduce subtle annoyances as it will swallow fields that might be mistyped but are not required etc.
